lib/Ethereum/RPC/Contract.pm

```diff
@@ -117,12 +121,20 @@ Return:
 =cut
 
 sub get_function_id {
-
-    my ($self, $function_string, @inputs) = @_;
+    my ($self, $function_string, $params_size) = @_;
+
+    my @inputs = @{$self->contract_decoded->{$function_string}};
+    my @selected_data = ();
+    if (scalar @inputs > 0) {
```

Comment:
`scalar` is optional here since the `if` already provides for boolean/scalar context.

Alternatively you can write it as `0 + @inputs` as well